src/extensions/nginx-app-protect/monitoring/processor/nap.go

```diff
@@ -483,6 +503,15 @@ func (f *NAPConfig) getViolations(logger *logrus.Entry) []*models.ViolationData
 				}
 				name = v.SpecificDesc
 				value = string(decodedBuffer)
+			} else if v.CookieLength != "" {
+				isB64Decoded = true
```

Comment:
question: similar question as above

Reply:
Based on all the examples we came across the assumption is correct. Param, URL, Cookie or Header is always b64 encoded.